### PR TITLE
Implement DNS-01 in ACME library

### PR DESCRIPTION
**Pull request by wteiken**

### CODE REVIEW

File: acme/acme/challenges.py [hunk 1]
```diff
@@ -216,6 +219,83 @@ def response_and_validation(self, account_key, *args, **kwargs):
...
+    def validation(self, account_key, **unused_kwargs):
+        """Generate validation.
+
+        :param JWK account_key:
+        :rtype: unicode
+
+        """
+        key_authorization = self.key_authorization(account_key)
+        # FIXME Once boulder response according to the spec this needs to be fixed
+        # return base64.b64encode(hashlib.sha256(key_authorization).digest())
```
Comment (alex) [inline acme/acme/challenges.py]: This should be `base64.urlsafe_b64encode`

**Comment by alex** [comment]
I have some overlapping code in #2052.

**Comment by wteiken** [comment]
No need for this branch then.

**Comment by alex** [comment]
@wteiken I think your branch is actually more complete and better tested than mine, I was suggesting you steal any code from it that looked useful :_

**Comment by wteiken** [comment]
Ok, I'll have another look then.   I wanted to pull the other request anyway to get the right email address in.

**Comment by wteiken** [comment]
Created new https://github.com/letsencrypt/letsencrypt/pull/2061

**Pull request closed without merging**